File: FragmentsApp/fragments/src/main/java/com/rjhartsoftware/fragments/TransactionsActivity.java
package com.rjhartsoftware.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class TransactionsActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        FragmentTransactions.activityCreated(this);
    }

    @Override
    protected void onPostResume() {
        super.onPostResume();
        FragmentTransactions.activityResumed(this);
    }

    @Override
    protected void onStart() {
        super.onStart();
        FragmentTransactions.activityStarted(this);
    }

    @Override
    protected void onPause() {
        FragmentTransactions.activityPaused(this);
        super.onPause();
    }

    @Override
    protected void onStop() {
        FragmentTransactions.activityStopped(this);
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        FragmentTransactions.activityDestroyed(this);
        super.onDestroy();
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        FragmentTransactions.activitySaved(this);
        super.onSaveInstanceState(outState);
    }

}

File: FragmentsApp/fragments/src/main/java/com/rjhartsoftware/fragments/TransactionsActivity.kt
package com.rjhartsoftware.fragments

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

open class TransactionsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        FragmentTransactions.activityCreated(this)
    }

    override fun onPostResume() {
        super.onPostResume()
        FragmentTransactions.activityResumed(this)
    }

    override fun onStart() {
        super.onStart()
        FragmentTransactions.activityStarted(this)
    }

    override fun onPause() {
        FragmentTransactions.activityPaused(this)
        super.onPause()
    }

    override fun onStop() {
        FragmentTransactions.activityStopped(this)
        super.onStop()
    }

    override fun onDestroy() {
        FragmentTransactions.activityDestroyed(this)
        super.onDestroy()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        FragmentTransactions.activitySaved(this)
        super.onSaveInstanceState(outState)
    }
}